src/main/java/build/buf/protovalidate/Format.java

```diff
@@ -34,6 +34,7 @@
 final class Format {
   private static final char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray();
   private static final char[] LOWER_HEX_ARRAY = "0123456789abcdef".toCharArray();
+  private static final DecimalFormat decimalFormat = new DecimalFormat("0.#########");
```

Comment:
DecimalFormat instances aren't thread safe - we should use a ThreadLocal to cache these (if we expect they'll be used a lot), add synchronization around them, or create them on demand:

> Decimal formats are generally not synchronized. It is recommended to create separate format instances for each thread. If multiple threads access a format concurrently, it must be synchronized externally.

https://docs.oracle.com/javase/8/docs/api/java/text/DecimalFormat.html